File: ver_py/T08.py
# https://leetcode.cn/problems/power-of-three/
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        ans = 1
        if n <= 0:
            return False
        elif n == 1:
            return True
        else:
            while ans < n:
                ans *= 3
            if ans == n:
                return True
            else:
                return False


if __name__ == "__main__":
    n = int(input())
    print(Solution().isPowerOfThree(n))

File: ver_cpp/T08.cpp
// https://leetcode.cn/problems/power-of-three/
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool isPowerOfThree(int n) {
        if (n <= 0) return false;
        if (n == 1) return true;
        long long ans = 1;
        while (ans < n) {
            ans *= 3;
            if (ans == n) return true;
        }
        return false;
    }
};

int main() {
    Solution s;
    long long n;
    cin >> n;
    cout << s.isPowerOfThree(n) << endl;
    return 0;
}
